src/Bicep.Core/TypeSystem/Az/AzResourceTypeProvider.cs

```diff
@@ -183,12 +189,14 @@ private static ObjectType SetBicepResourceProperties(ObjectType objectType, Reso
                 {
                     properties = properties.SetItem(LanguageConstants.ResourceLocationPropertyName, new TypeProperty(LanguageConstants.ResourceLocationPropertyName, locationTypeProperty.TypeReference, locationTypeProperty.Flags | TypePropertyFlags.DeployTimeConstant));
                 }
-                
-                // we only support scope for extension resources (or resources where the scope is unknown and thus may be an extension resource)
-                if (validParentScopes.HasFlag(ResourceScope.Resource))
+
+                // we only support scope in these cases:
+                // 1. extension resources (or resources where the scope is unknown and thus may be an extension resource)
+                // 2. Tenant resources
+                ResourceScope tenantOrExtensionResourceScope = validParentScopes & (ResourceScope.Resource | ResourceScope.Tenant);
+                if (tenantOrExtensionResourceScope != 0)
```

Comment:
Although this is obviously pretty straightforward logic, worth moving it into `ScopeHelper`? I've been trying to keep all of the logic to enforce ARM-specific scope validations in that place.

Reply:
Sure!

Reply:
Done